src/App.tsx

```diff
@@ -1,60 +1,150 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
 
 import classNames from 'classnames';
-import { PostsList } from './components/PostsList';
-import { PostDetails } from './components/PostDetails';
-import { UserSelector } from './components/UserSelector';
+import { PostsList } from './components/PostsList/PostsList';
+import { PostDetails } from './components/PostDetails/PostDetails';
+import { UserSelector } from './components/UserSelector/UserSelector';
 import { Loader } from './components/Loader';
+import { getComments, getPosts, getUsers } from './api/posts';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { Comment } from './types/Comment';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [isLoadingUsers, setIsLoadingUsers] = useState<boolean>(false);
+  const [isErrorPosts, setIsErrorPosts] = useState<boolean>(false);
+
+  const [posts, setPosts] = useState<Post[] | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isErrorComments, setIsErrorComments] = useState<boolean>(false);
+  const [isLoadingComments, setIsLoadingComments] = useState<boolean>(false);
+  const [isFormShown, setIsFormShown] = useState<boolean>(false);
+
+  useEffect(() => {
+    getUsers()
+      .then(setUsers)
+      .catch(() => setIsErrorPosts(true));
+  }, []);
+
+  useEffect(() => {
+    if (selectedUser) {
+      setIsLoadingUsers(true);
+
+      getPosts(selectedUser.id)
+        .then((resolve) => {
+          setPosts(resolve);
+        })
+        .catch(() => {
+          setIsErrorPosts(true);
+        })
+        .finally(() => {
+          setIsLoadingUsers(false);
+        });
+    }
+  }, [selectedUser]);
+
+  useEffect(() => {
+    setIsErrorComments(false);
+    setIsLoadingComments(true);
+    if (selectedPost) {
+      getComments(selectedPost.id)
+        .then((resolve) => {
+          setComments(resolve);
+        })
+        .catch((error: PromiseRejectedResult) => {
+          setIsErrorComments(true);
+          throw new Error(error.status);
```

Comment:
do not throw it - in this case, the page will crash (a thrown error will not be handled at all)
you already handled an error in the way is acceptable for us (just show comments error)